### PR TITLE
gitserver: Simplify gitserver_repos updates

### PR DESCRIPTION
**Pull request by ryanslade**
Simplify updates to the `gitserver_repos` table by taking repo name instead
of id and looking up the id as part of the insert.

As a side effect this means that we are no longer explicitly looking up a repo
which avoids our security logging code.

### CODE REVIEW

File: cmd/gitserver/main.go [hunk 1]
```diff
@@ -237,9 +238,13 @@ func getPercent(p int) (int, error) {
 // getStores initializes a connection to the database and returns RepoStore and
 // ExternalServiceStore.
 func getDB() (dbutil.DB, error) {
+
 	//
 	// START FLAILING
 
+	// TODO: Remove this hack once we're confident we're using contexts with internal
+	// actors everywhere
```
Comment (unknwon) [inline cmd/gitserver/main.go:246]: This is required until we change the way we write the [AuthzQueryConds](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/repos_perm.go?utm_product_name=GoLand&utm_product_version=GoLand#L22:54). We can potentially check internal actor first before trying to list out authz providers.

File: cmd/gitserver/server/server.go [hunk 2]
```diff
@@ -225,6 +226,9 @@ var longGitCommandTimeout = time.Hour
 // Handler returns the http.Handler that should be used to serve requests.
 func (s *Server) Handler() http.Handler {
 	s.ctx, s.cancel = context.WithCancel(context.Background())
+	// The server context should use an internal actor as it needs access to all
+	// repos.
+	s.ctx = actor.WithInternalActor(s.ctx)
```
Comment (unknwon) [inline cmd/gitserver/server/server.go:231]: Is it safe to assume we always need the internal actor here? This is not a background process and gitserver may serve user requests.

File: cmd/gitserver/main.go (hunk 1 above)
Comment (ryanslade) [reply cmd/gitserver/main.go:246]: Isn't this a hack to trick it into bypassing authz? I thought the proper way to do it is to use an internal actor

File: cmd/gitserver/server/server.go (hunk 2 above)
Comment (ryanslade) [reply cmd/gitserver/server/server.go:231]: Well, the current code is already bypassing authz for all queries right? 

My assumption is that as with our other services, frontend should perform authz checks and internally our "backend" services can skip authz.

File: cmd/gitserver/main.go (hunk 1 above)
Comment (ryanslade) [reply cmd/gitserver/main.go:246]: We should find one way of doing this as clearing out the providers effectively removes all authz checking in gitserver

Comment (unknwon) [reply cmd/gitserver/main.go:246]: >Isn't this a hack to trick it into bypassing authz?

No, it is the trick to unblock calls of `authz.GetProviders()`, it is waiting for a call of `authz.SetProviders`, which marks authz ready:

https://github.com/sourcegraph/sourcegraph/blob/11e64aa750195005c6158f5e716d5930344d587f/internal/authz/register.go#L37-L39

File: cmd/gitserver/server/server.go (hunk 2 above)
Comment (unknwon) [reply cmd/gitserver/server/server.go:231]: > Well, the current code is already bypassing authz for all queries right?

Without an internal actor, I'd say no.

> My assumption is that as with our other services, frontend should perform authz checks and internally our "backend" services can skip authz.

Genereally agree, though.

**Comment by ryanslade** [comment]
@unknwon I ended up repurposing this PR to improve the way we update `gitserver_repos`.

**Review by unknwon — APPROVED** [review]

**Pull request merged**